core/src/main/java/org/infinispan/remoting/transport/jgroups/JGroupsAddressCache.java

```diff
@@ -0,0 +1,42 @@
+package org.infinispan.remoting.transport.jgroups;
+
+import org.infinispan.commons.equivalence.AnyEquivalence;
+import org.infinispan.commons.util.concurrent.jdk8backported.EquivalentConcurrentHashMapV8;
+import org.jgroups.Address;
+import org.jgroups.util.ExtendedUUID;
+import org.jgroups.util.UUID;
+
+/**
+ * Cache JGroupsAddress instances
+ *
+ * @author Dan Berindei
+ * @since 7.0
+ */
+public class JGroupsAddressCache {
+   private static final EquivalentConcurrentHashMapV8<Address, JGroupsAddress> addressCache =
+         new EquivalentConcurrentHashMapV8<>(AnyEquivalence.getInstance(), AnyEquivalence.getInstance());
+
+   static org.infinispan.remoting.transport.Address fromJGroupsAddress(final org.jgroups.Address addr) {
```

Comment:
Also I wonder we could have 2 methods here, 1 that takes a regular org.jgroups.Address and another that takes a org.jgroups.util.ExtendedUUID and returns the proper type.  At least then the cast is isolated in 1 spot and not every caller has to cast like we have atm.